coala_quickstart/coala_quickstart.py

```diff
@@ -90,15 +91,24 @@ def main():
 
     extracted_information = collect_info(project_dir)
 
-    relevant_bears = filter_relevant_bears(
-        used_languages, printer, arg_parser, extracted_information)
-
-    print_relevant_bears(printer, relevant_bears)
+    # OrderedDict used for print_relevant_bears to first print unusable bears
+    relevant_bears = OrderedDict(
+            [('unusable', {}),
+             ('usable',
+                 filter_relevant_bears(used_languages,
+                                       printer,
+                                       arg_parser,
+                                       extracted_information))])
 
     if args.non_interactive and not args.incomplete_sections:
-        unusable_bears = get_non_optional_settings_bears(relevant_bears)
+        unusable_bears = get_non_optional_settings_bears(
+            relevant_bears['usable'])
         remove_unusable_bears(relevant_bears, unusable_bears)
-        print_relevant_bears(printer, relevant_bears, 'usable')
+
+    print_relevant_bears(printer, relevant_bears)
+
+    # Drop unusable bears
+    relevant_bears = relevant_bears['usable']
```

Comment:
Isn't the `remove_unusable_bears` method doing this?

Reply:
https://github.com/coala/coala-quickstart/pull/227/files#diff-e09a499ccad128bbc49bfc6022ca886eR106

Reply:
I changed `remove_unusable_bears` to also add the `unusable` bears to the OrderdDict, but I'm using that only for printing purposes. Here the printing has ended, so I drop `unusable` bears from the dict.

Reply:
> I changed remove_unusable_bears to also add the unusable bears to the OrderdDict

doesn't that contradict with the function name? `remove_unusable_bears` is adding unusable bears to `OrdereDict`